wasm/ro/chapter_3.yaml

```
@@ -0,0 +1,263 @@
+
+
+    ```rust
+
+    let log_handle = register_function("
+      (param_a, param_b) => {
+        // apelează cumva console_log  
+      }");
+
+    let msg = "hello world";
+
+
+    js_invoke_with_2_params( log_handle,msg.as_ptr() as f64, 
+        msg.len() as f64 );
+    ```
+
+
+    Veți observa că există o problemă cu funcția noastră JavaScript. Are
+    începutul și sfârșitul șirului utf-8, dar nu și memoria acestuia!
+- title: Contextul invocării dinamice
+  code: "https://webassembly.studio/?embed&f=70lhg7nwda9"
+  content_markdown: >
+    Când funcțiile noastre invocate dinamic sunt apelate, acestea trebuie să
+    aibă acces la memoria modulului. Putem plasa
+    
+    resurse importante unui obiect de context astfel încât funcția noastră să
+    aibă toate instrumentele disponibile pentru a face ceea ce trebuie.
+
+
+    ```rust
+
+    let log_handle = register_function("
+      (context, msgStart, msgEnd) => {
+        let msg = context.getUtf8FromMemory(msgStart,msgEnd);
+        console.log(msg); 
+      }");
+
+    let msg = "hello world";
+
+
+    js_invoke_with_2_params( log_handle,msg.as_ptr() as f64, 
+      msg.len() as f64);
+    ```
+
+
+    Studiați exemplul pentru implementarea completă.
+- title: Returnarea referințelor
+  content_markdown: >
+    Când vrem să returnăm o referință la un obiect, întâlnim o problemă la funcțiile
+    dinamice. WebAssembly poate să trimită înainte și înapoi doar numere!
+
+    Pentru e trece peste acest obstacol, trebuie să stabilim o convenție de a folosi
+    un identificator numeric pentru a reprezenta
+
+    o referință către obiectul nostru în JavaScript.
+
+
+    ```rust
+
+    let query_selector_handle = register_function("
+      (context, selectorStart, selectorEnd) => {
+        let selector = context.getUtf8FromMemory(
+          selectorStart,selectorEnd);
+        let domEl = document.querySelector(selector);
+        let objHandle = context.storeObject(domEl);
+        return objHandle;
+      }");
+
+    let selector = "#fancy-button";
+
+
+    let dom_element_handle = js_invoke_with_2_params(
+        log_handle,selector.as_ptr() as f64, 
+        selector.len() as f64 );
+    ```
+- title: Alocatoare
+  code: "https://webassembly.studio/?embed&f=mz2gu9y29tn"
+  content_markdown: >
+    După cum putem vedea, pentrz a comunica corect obiecte, avem nevoie de un fel de
+    mecanism de stocare pentru obiecte:
+
+
+    * `storeObject(object) -> f64` - Stochează un obiect și returnează un identificator
+      numeric.
+
+    * `getObject(f64) -> object` - Obține o referință la un obiect prin identificatorul
+      său numeric.
+
+    * `releaseObject(f64) -> object` - Eliberează o referință la un obiect din
+      spațiul de stocare, permițând eliberarea memoriei sale.
+
+    
+    Implementarea unei astfel de funcționalități se numește *alocator*. Acesta este
+    dincolo de sfera acestui tutorial. Ne-am putea imagina o implementae foarte simplă a
```

```suggestion
    dincolo de sfera acestui tutorial. Ne-am putea imagina o implementare foarte simplă a
```
Typo: implementare in loc de implementae